pytorch_toolkit/object_detection/oteod/oteod/misc.py

```diff
@@ -0,0 +1,120 @@
+# pylint: disable=W0702,W1203,R0913
+
+import hashlib
+import json
+import os
+import signal
+import subprocess
+import sys
+import time
+from queue import Queue, Empty
+from threading import Thread
+
+from oteod import MMDETECTION_TOOLS
+
+
+class NonBlockingStreamReader:
+
+    def __init__(self, stream):
+        self.stream = stream
+        self.queue = Queue()
+
+        def populate_queue(stream, queue):
+            while True:
+                line = stream.readline()
+                if line:
+                    queue.put(line)
```

Comment:
May be it's better to add `wait(?)` here in `else` section to reduce cpu loading?

Reply:
added